tests/test_machine_only_rules.py

```diff
@@ -21,58 +31,94 @@ def main():
         guide_dir = os.path.abspath(
             os.path.join(product_dir, product_yaml['benchmark_root']))
         additional_content_directories = product_yaml.get("additional_content_directories", [])
-        add_content_dirs = [os.path.abspath(os.path.join(product_dir, rd)) for rd in additional_content_directories]
+        add_content_dirs = [os.path.abspath(
+            os.path.join(product_dir, rd)) for rd in additional_content_directories]
         if not check_product(args.build_dir, product, [guide_dir] + add_content_dirs):
             test_result = 1
     if test_result:
         sys.exit(1)
 
+
 def check_product(build_dir, product, rules_dirs):
     input_groups, input_rules = scan_rules_groups(rules_dirs, False)
     ds_path = os.path.join(build_dir, "ssg-" + product + "-ds.xml")
-    if not check_ds(ds_path, "Group", input_groups):
-        return False
-    if not check_ds(ds_path, "Rule", input_rules):
+    if machine_platform_missing_in_rules(ds_path, input_rules):
         return False
     return True
 
 
-def check_ds(ds_path, what, input_elems):
+def shorten_id(full_id):
+    id_prefix = "xccdf_org.ssgproject.content_rule_"
+    return full_id.replace(id_prefix, "")
+
+
+def get_only_elements_to_check_from_benchmark(benchmark, element_query, short_ids_to_check):
+    elements = []
+    for elem in benchmark.findall(element_query):
+        elem_short_id = shorten_id(elem.get("id"))
+        if elem_short_id in short_ids_to_check:
+            elements.append(elem)
+    return elements
+
+
+def get_element_fix_text_by_system(element):
+    all_fixes = element.findall("{%s}fix" % ssg.constants.XCCDF12_NS)
+    fixes_by_system = dict()
+    for f in all_fixes:
+        system = f.get("system")
+        fixes_by_system[system] = "".join(f.itertext())
+    return fixes_by_system
+
+
+def machine_platform_missing_in_rules(ds_path, short_ids_to_check):
+    machine_platform_missing = False
+
     try:
         tree = ET.parse(ds_path)
-    except IOError as e:
+    except IOError:
```

Comment:
If the data stream doesn't exist, it doesn't make sense to scan and load compiled rules for that product.